roosevelt-less.js

```diff
@@ -1,62 +1,58 @@
 const fs = require('fs')
+const path = require('path')
 const LessPluginCleanCSS = require('less-plugin-clean-css')
 const less = require('less')
 
 module.exports = {
   versionCode: function (app) {
-    return '@' + app.get('params').versionedCssFile.varName + ': \'' + app.get('appVersion') + '\';\n'
+    return `@${app.get('params').versionedCssFile.varName}: '${app.get('appVersion')}';\n`
   },
 
-  parse: function (app, fileName, callback) {
-    let defaultCompress = true
-
-    // LESS render options
-    let options = {
-      filename: app.get('cssPath') + fileName,
-      paths: app.get('cssPath')
-    }
-
-    // Clean-css options
-    let opts
-
-    // LESS clean-css plugin
-    let cleanCSSPlugin
-
-    // disable minify if noMinify param is present in roosevelt
-    if (app.get('params').noMinify) {
-      defaultCompress = false
-      app.get('params').cssCompiler.params.advanced = defaultCompress
-      app.get('params').cssCompiler.params.aggressiveMerging = defaultCompress
-    }
-
-    // Enable minification and compilation with less-plugin-clean-css
-    if (defaultCompress) {
-      opts = app.get('params').cssCompiler.params || {
-        advanced: defaultCompress,
-        aggressiveMerging: defaultCompress
+  parse: function (app, fileName) {
+    return new Promise((resolve, reject) => {
+      const params = app.get('params').cssCompiler
+
+      // LESS render options
+      let options = {
+        paths: app.get('cssPath'),
+        filename: path.basename(fileName)
       }
 
-      // Re-interpret 'compress' param into valid clean-css options
-      if (opts.hasOwnProperty('compress')) {
-        opts.advanced = true
-        opts.aggressiveMerging = true
-        delete opts.compress
+      let lessInput = fs.readFileSync(path.join(app.get('cssPath'), fileName), 'utf8')
+
+      // Clean-css options
+      let opts
+
+      // LESS clean-css plugin
+      let cleanCSSPlugin
+
+      if (params.sourceMap) {
```

Comment:
This needs a little tweaking. Source map is accessed at `params.params.sourceMap` since you have a key `params` in the object `cssCompiler`. The `object.assign` doesn't work exactly as you would like either. It does copy the value but you still need to create and preserve the top level key `sourceMap`.

For example given what you have:

With the pacakge.json param:

```json
"cssCompiler": {
  "nodeModule": "roosevelt-less",
  "params": {
    "cleanCSS": {
      "advanced": true,
      "aggressiveMerging": true
    },
    "sourceMap": {

    }
  }
}
```

Will give the less compiler: 

```
{ paths: 'C:\\Users\\alallier\\Documents\\cybertime\\statics\\css\\',
  filename: 'timeline.less' }
```

And the package.json param:

```json
"cssCompiler": {
  "nodeModule": "roosevelt-less",
  "params": {
    "cleanCSS": {
      "advanced": true,
      "aggressiveMerging": true
    },
    "sourceMap": {
      "sourceMapFileInline": true
    }
  }
}
```

Will give:

```
{ paths: 'C:\\Users\\alallier\\Documents\\cybertime\\statics\\css\\',
  filename: 'timeline.less',
  sourceMapFileInline: true }
```

When it needs to be

```
{ paths: 'C:\\Users\\alallier\\Documents\\cybertime\\statics\\css\\',
  filename: 'timeline.less',
  sourceMap: {} }
```

or

```
{ paths: 'C:\\Users\\alallier\\Documents\\cybertime\\statics\\css\\',
  filename: 'timeline.less',
  sourceMap: { sourceMapFileInline: true } }
```

It's also worth noting `sourceParam: {}` (empty object) still does source mapping, if you don't want source mapping at all the key (sourceParam) needs to not exist in the options object. Which might make you have to tweak your default.

I was able to get it working with:

```js
if (params.params.sourceMap) {
  options.sourceMap = params.params.sourceMap;
}
```

You might find a cleaner way to do it.